### ISSUE TITLE
Some gem type properties are not going into `gem.type`

### ISSUE THREAD

**Issue by trackpete** [trackpete/exiletools-indexer#52]
There are a few type properties for some gems that aren't going into `gem.type` including:

* `properties.Gem.Spell`
* `properties.Gem.Support`

An example query to find these:

```
GET poe/item/_search
{
"query": {
  "filtered": {
    "query": {
      "term": {
        "properties.Gem.Spell": {
          "value": true
        }
      }

      }
    }
  }
}
}
```

An example m5sum for spell:

`56a94af8f45f24edb0cd362482a45682`

An example of one that works:

`4f045b966e765a8eeaf11b4683b36fb2`

**Comment by trackpete** [comment]
Let's find all the bad ones:

```
GET poe/item/_search
{
"query": {
  "filtered": {
    "query": {
      "term": {
        "properties.Gem.Spell": {
          "value": true
        }
      }

      }
    }
  },
  "aggs": {
    "NAME": {
      "terms": {
        "field": "info.fullName",
        "size": 100
      }
    }
  },
  size:0
}
```

Looks like this problem is with `Portal` and `Detonate Mines`

Let's do the same for Support:

```
GET poe/item/_search
{
"query": {
  "filtered": {
    "query": {
      "term": {
        "properties.Gem.Support": {
          "value": true
        }
      }

      }
    }
  },
  "aggs": {
    "NAME": {
      "terms": {
        "field": "info.fullName",
        "size": 100
      }
    }
  },
  size:0
}
```

Bunch of ones here. Basically any item with only one `type` isn't detected properly. Code problem?

**Comment by trackpete** [comment]
Shit, yep, from `sub.formatJSON.pl`:

```
        if ($property =~ /\,/) {
          my @props = split(/\,/, $property);
          foreach $prop (@props) {
            $prop =~ s/^\s+//g;
            $item{properties}{$item{attributes}{baseItemType}}{type}{$prop} = \1;
          }
```

How the f' do we fix these?

Is it possible that for Gems, `type` properties are the *only properties without values?*

For example:

```
   ],
   "properties" : [
      {
         "displayMode" : 0,
         "name" : "Support",
         "values" : []
      },
```

I see no other booleans in the mappings. So, modify `sub.formatJSON.pl` to assume the property is a `type` if it's a boolean? Will that always work for Gems? Who knows. Let's at least quickly look at the new gems.

**Comment by trackpete** [comment]
Anyone reading this must think I'm crazy for constantly talking to myself in my issues. My issues got issues.

**Comment by trackpete** [comment]
Okay, fixed by assuming any boolean gem property is a type. Hopefully that stays the case.

**Comment by thirdy** [comment]
> Anyone reading this must think I'm crazy for constantly talking to myself
in my issues. My issues got issues.

haha

well I'm subscribed to your github project, so I get emails for any updates.

anyway, it's good to learn from your logs working on these issues. cheers!


On Fri, Dec 11, 2015 at 5:05 AM, Pete Waterman <notifications@github.com>
wrote:

> Closed #52 <https://github.com/trackpete/exiletools-indexer/issues/52>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/trackpete/exiletools-indexer/issues/52#event-488673971>
> .
>



-- 
--
http://sites.google.com/site/thirdyderivera/
